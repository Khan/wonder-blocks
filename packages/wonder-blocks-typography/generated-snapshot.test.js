// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests
// edit packages/wonder-blocks-typography/docs.md and run `npm run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";
import BodyMonospace from "./components/body-monospace.js";
import BodySerifBlock from "./components/body-serif-block.js";
import BodySerif from "./components/body-serif.js";
import Body from "./components/body.js";
import Caption from "./components/caption.js";
import Footnote from "./components/footnote.js";
import HeadingLarge from "./components/heading-large.js";
import HeadingMedium from "./components/heading-medium.js";
import HeadingSmall from "./components/heading-small.js";
import HeadingXSmall from "./components/heading-xsmall.js";
import LabelLarge from "./components/label-large.js";
import LabelMedium from "./components/label-medium.js";
import LabelSmall from "./components/label-small.js";
import LabelXSmall from "./components/label-xsmall.js";
import Tagline from "./components/tagline.js";
import Title from "./components/title.js";

describe("wonder-blocks-typography", () => {
    it("example 1", () => {
        const {View} = require("wonder-blocks-core");
        
        const example = <View>
            <Title>Title</Title>
            <HeadingLarge>HeadingLarge</HeadingLarge>
            <HeadingMedium>HeadingMedium</HeadingMedium>
            <HeadingSmall>HeadingSmall</HeadingSmall>
            <HeadingXSmall>HeadingXSmall</HeadingXSmall>
            <BodySerifBlock>BodySerifBlock</BodySerifBlock>
            <BodySerif>BodySerif</BodySerif>
            <BodyMonospace>BodyMonospace</BodyMonospace>
            <Body>Body</Body>
            <LabelLarge>LabelLarge</LabelLarge>
            <LabelMedium>LabelMedium</LabelMedium>
            <LabelSmall>LabelSmall</LabelSmall>
            <LabelXSmall>LabelXSmall</LabelXSmall>
            <Tagline>Tagline</Tagline>
            <Caption>Caption</Caption>
            <Footnote>Footnote</Footnote>
        </View>
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 2", () => {
        const Color = require("wonder-blocks-color").default;
        const {StyleSheet} = require("aphrodite");
        
        const styles = StyleSheet.create({
            blueText: {
                color: Color.blue,
            },
        });
        
        const example = <Title style={styles.blueText}>Blue Title</Title>
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
