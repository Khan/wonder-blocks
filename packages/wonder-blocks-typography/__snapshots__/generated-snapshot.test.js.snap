// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-typography example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <h1
    className=""
    id="example-Title"
    style={
      Object {
        "@media (maxWidth: 1023px)": Object {
          "fontSize": 28,
          "lineHeight": "32px",
        },
        "@media (minWidth: 1024px)": Object {
          "fontSize": 36,
          "lineHeight": "40px",
        },
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontWeight": 900,
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    Title
  </h1>
  <h2
    className=""
    id="example-HeadingLarge"
    style={
      Object {
        "@media (maxWidth: 1023px)": Object {
          "fontSize": 24,
          "lineHeight": "28px",
        },
        "@media (minWidth: 1024px)": Object {
          "fontSize": 28,
          "lineHeight": "32px",
        },
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontWeight": 700,
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    HeadingLarge
  </h2>
  <h3
    className=""
    id="example-HeadingMedium"
    style={
      Object {
        "@media (maxWidth: 1023px)": Object {
          "fontSize": 22,
          "lineHeight": "26px",
        },
        "@media (minWidth: 1024px)": Object {
          "fontSize": 24,
          "lineHeight": "28px",
        },
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontWeight": 700,
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    HeadingMedium
  </h3>
  <h4
    className=""
    id="example-HeadingSmall"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 20,
        "fontWeight": 700,
        "lineHeight": "24px",
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    HeadingSmall
  </h4>
  <h4
    className=""
    id="example-HeadingXSmall"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 12,
        "fontWeight": 700,
        "letterSpacing": 0.6,
        "lineHeight": "16px",
        "marginBottom": 0,
        "marginTop": 0,
        "textTransform": "uppercase",
      }
    }
  >
    HeadingXSmall
  </h4>
  <span
    className=""
    id="example-BodySerifBlock"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "serif",
        "fontSize": 22,
        "fontWeight": 400,
        "lineHeight": "28px",
      }
    }
  >
    BodySerifBlock
  </span>
  <span
    className=""
    id="example-BodySerif"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "serif",
        "fontSize": 18,
        "fontWeight": 400,
        "lineHeight": "22px",
      }
    }
  >
    BodySerif
  </span>
  <span
    className=""
    id="example-BodyMonospace"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Inconsolata, monospace",
        "fontSize": 17,
        "fontWeight": 400,
        "lineHeight": "22px",
      }
    }
  >
    BodyMonospace
  </span>
  <span
    className=""
    id="example-Body"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": "22px",
      }
    }
  >
    Body
  </span>
  <span
    className=""
    id="example-LabelLarge"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 16,
        "fontWeight": 700,
        "lineHeight": "20px",
      }
    }
  >
    LabelLarge
  </span>
  <span
    className=""
    id="example-LabelMedium"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": "20px",
      }
    }
  >
    LabelMedium
  </span>
  <span
    className=""
    id="example-LabelSmall"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 14,
        "fontWeight": 400,
        "lineHeight": "18px",
      }
    }
  >
    LabelSmall
  </span>
  <span
    className=""
    id="example-LabelXSmall"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 12,
        "fontWeight": 400,
        "lineHeight": "16px",
      }
    }
  >
    LabelXSmall
  </span>
  <span
    className=""
    id="example-Tagline"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 20,
        "fontWeight": 400,
        "lineHeight": "24px",
      }
    }
  >
    Tagline
  </span>
  <span
    className=""
    id="example-Caption"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 14,
        "fontWeight": 400,
        "lineHeight": "20px",
      }
    }
  >
    Caption
  </span>
  <span
    className=""
    id="example-Footnote"
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 12,
        "fontWeight": 400,
        "lineHeight": "18px",
      }
    }
  >
    Footnote
  </span>
</div>
`;

exports[`wonder-blocks-typography example 2 1`] = `
<h1
  className=""
  style={
    Object {
      "@media (maxWidth: 1023px)": Object {
        "fontSize": 28,
        "lineHeight": "32px",
      },
      "@media (minWidth: 1024px)": Object {
        "fontSize": 36,
        "lineHeight": "40px",
      },
      "MozOsxFontSmoothing": "grayscale",
      "WebkitFontSmoothing": "antialiased",
      "color": "#1865f2",
      "display": "block",
      "fontFamily": "Lato, sans-serif",
      "fontWeight": 900,
      "marginBottom": 0,
      "marginTop": 0,
    }
  }
>
  Blue Title
</h1>
`;

exports[`wonder-blocks-typography example 3 1`] = `
<h1
  aria-label="Accessible Title"
  className=""
  style={
    Object {
      "@media (maxWidth: 1023px)": Object {
        "fontSize": 28,
        "lineHeight": "32px",
      },
      "@media (minWidth: 1024px)": Object {
        "fontSize": 36,
        "lineHeight": "40px",
      },
      "MozOsxFontSmoothing": "grayscale",
      "WebkitFontSmoothing": "antialiased",
      "display": "block",
      "fontFamily": "Lato, sans-serif",
      "fontWeight": 900,
      "marginBottom": 0,
      "marginTop": 0,
    }
  }
>
  Title
</h1>
`;
